File: app/src/main/java/com/example/linkopener/MainActivity.java
package com.example.linkopener;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.example.linkopener.history.HistoryActivity;

/**
 * Launcher activity
 */
public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    /**
     * Launch activity and expect the URL as the result
     */
    ActivityResultLauncher<Intent> getURL = registerForActivityResult(
            new ActivityResultContracts.StartActivityForResult(),
            result -> {
                switch (result.getResultCode()) {
                    case Activity.RESULT_OK:
                        Intent data = result.getData();
                        if (data != null) {
                            OpenLinkDialogFragment dialog = new OpenLinkDialogFragment(data.getData());
                            dialog.show(getSupportFragmentManager(), "Tag");
                        }
                        break;
                    case Activity.RESULT_CANCELED:
                        break;
                    default:
                        Toast.makeText(this, getString(R.string.link_handling_error), Toast.LENGTH_SHORT).show();
                }
            }
    );

    /**
     * Open link enterer page callback
     */
    public void switchToLinkEnterer(View view) {
        Intent openLinkEnterer = new Intent(this, LinkEntererActivity.class);
        getURL.launch(openLinkEnterer);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.main_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == R.id.action_history) {
            Intent history = new Intent(this, HistoryActivity.class);
            startActivity(history);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/example/vkalbums/MainActivity.kt
package com.example.vkalbums

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

/**
 * Activity with user input
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val userIdEdit: EditText = findViewById(R.id.user_id_edit)
        val getAlbumsButton: Button = findViewById(R.id.get_albums_button)
        getAlbumsButton.setOnClickListener {
            val userId = userIdEdit.text.toString()
            if (userId.isNotEmpty()) {
                val intent = Intent(this@MainActivity, AlbumsActivity::class.java)
                intent.putExtra("user_id", userId)
                startActivity(intent)
            }
        }
    }
}